xonsh/lexer.py

```diff
@@ -33,53 +327,26 @@ def __init__(self, errfunc=lambda e, l, c: print(e)):
         self.errfunc = errfunc
         self.fname = ''
         self.last = None
-        self.lexer = None
-        self.indent = ''
-        self.in_py_mode = [True]
 
     def build(self, **kwargs):
         """Part of the PLY lexer API."""
-        self.lexer = lex.lex(object=self, **kwargs)
-        self.reset()
+        pass
 
     def reset(self):
-        self.lexer.lineno = 1
-        self.indent = ''
-        self.last = None
-        self.in_py_mode = [True]
-        self.in_parens = [False]
-
-    @property
-    def lineno(self):
-        if self.lexer is not None:
-            return self.lexer.lineno
-
-    @lineno.setter
-    def lineno(self, value):
-        if self.lexer is not None:
-            self.lexer.lineno = value
+        pass
 
     def input(self, s):
         """Calls the lexer on the string s."""
-        self.lexer.input(s)
+        s = re.sub(r'#.*', '', s)
```

Comment:
Can't we just use the functions above to ignore comments?  Also, regexes like this should be precompiled and not compiled on each function call.